tests/chainerx_tests/unit_tests/routines_tests/test_connection.py

```diff
@@ -82,6 +107,10 @@ def setup(self):
         if device.backend.name == 'cuda' and self.cover_all:
             pytest.skip('cudnn does not support cover_all')
 
+        # Skip NCWH test except in CUDA
+        if device.backend.name != 'cuda' and self.data_format == 'nhwc':
+            pytest.skip('Skip NCWH test except in CUDA')
```

Comment:
Is this skip necessary?
Furthermore, to skip a test, you should raise `unittest.SkipTest` instead of `pytest.skip` due to a convention.

Reply:
Yes, this skip is necessary because the size of the input array tends to be so large.

Reply:
Why do you need large arrays?

Reply:
It is because `cudnnFindConvolution***AlgorithmEx` chooses an algorithm which uses TensorCore only when using large arrays.

Reply:
I'll add these comment.

Reply:
(As we discussed offline,) we decided to split the test class.